src/actions/notificationActions.js

```diff
@@ -1,18 +1,21 @@
 import { notificationTypes } from "./types";
 
-export const addSnackbar = (notification) => (dispatch) => {
-    dispatch({
+export const addSnackbar = (notification) => {
+
+    if (typeof notification.message !== "string" || notification.message.length === 0) {
+        throw new Error("Notification must have a message field");
+    }
+
+    return {
         type: notificationTypes.ADD_SNACKBAR,
         notification: {
-            key: new Date().getTime() + Math.random(),
+            key: Date.now() + Math.random(),
             ...notification,
         },
-    });
+    };
 };
 
-export const removeSnackbar = (key) => (dispatch) => {
-    dispatch({
-        type: notificationTypes.REMOVE_SNACKBAR,
-        key,
-    });
-};
+export const removeSnackbar = (key) => ({
+    type: notificationTypes.REMOVE_SNACKBAR,
+    key,
```

Comment:
There is no standard regarding Redux Actions structure (https://redux.js.org/basics/actions), check their example. As long as the reducer matches the object fields, it should be fine. The only action field that is "mandatory" is `type`

Reply:
Right, but stuff like `redux-actions` and similar make use of `payload`. If we eventually change to it, we would have to switch everything, and having stuff nested in the payload looks like a cleaner structure to me, you would always expect the `type` and `payload` keys, instead of varying ones.

Reply:
For me, it just seems like overcomplicating what doesn't need to be overcomplicated. Seems we would be optimizing just for a package we probably won't use, and we should keep actions as they are: Objects with at least a `type ` property, being the additional properties defined by the dev depending on the needs of the action.

Reply:
Sure, I can roll with that - it doesn't seem like it will make any big difference in the end, like you mentioned